cocos/core/renderer/scene/light.ts

```diff
@@ -216,7 +217,12 @@ export class Light {
     public destroy () {
         this._name = null;
         this._node = null;
+        this._recycle();
     }
 
     public update () {}
+
+    protected _recycle () {
+        legacyCC.director.root.lightPool(this).free();
```

Comment:
Add `Root.recycleLight` API please, it's allocated in root, so it should also be recycled there. Be very cautious about this kind of thing